service/src/vsids.rs

```diff
@@ -68,8 +65,8 @@ pub enum GraphOp {
 
 #[derive(Clone, Debug)]
 pub struct VSIDSManager {
-  weights: HashMap<Edge, Weight>,
-  max_indices: HashMap<SourceKey, Weight>,
+  weights: HashMap<(String, NodeId, NodeId), Weight>,
```

Comment:
As was pointed out earlier, you do not need to store a copy of the graph weights. You can reuse the actual graph weights from the real graph. For that, you can use that object (the actual graph) as an immutable reference argument in VSIDS manager operations such as `update_weight`.